src/PrintReceipt.ts

```diff
@@ -1,12 +1,125 @@
 import {loadAllItems, loadPromotions} from './Dependencies'
+export interface Quantity{
+  value: number;
+  quantifier: string
+}
+
+interface ReceiptItem{
+  name: string;
+  quantity: Quantity;
+  unitPrice: number;
+  subtotal: number;
+  discountedPrice: number
+}
+
+interface Tag{
+  barcode: string;
+  quantity: number
+}
 
 export function printReceipt(tags: string[]): string {
-  return `***<store earning no money>Receipt ***
-Name：Sprite，Quantity：5 bottles，Unit：3.00(yuan)，Subtotal：12.00(yuan)
-Name：Litchi，Quantity：2.5 pounds，Unit：15.00(yuan)，Subtotal：37.50(yuan)
-Name：Instant Noodles，Quantity：3 bags，Unit：4.50(yuan)，Subtotal：9.00(yuan)
-----------------------
-Total：58.50(yuan)
-Discounted prices：7.50(yuan)
-**********************`
+
+  const parsedTags = parseTags(tags)
+  if(parsedTags === null){
+    return 'Tags are invalid, please check it!'
+  }
+  const receiptItems = generateReceiptItems(parsedTags)
+  const receipt = renderReceipt(receiptItems)
+  return receipt
+}
+
+function isTagValid(parsedTag: Tag): boolean{
+
+  return true
+  // if(parsedTag.barcode in )
+  // barcode: tag.slice(0,10),
+  // quantity: quantity
+}
+
+function parseOneTag(tag: string): Tag | null{
+  let parsedTag:Tag
+  if(tag.includes('-')){
+    const splitTag:string[] = tag.split('-')
+    parsedTag = {
+      barcode: splitTag[0],
+      quantity: parseInt(splitTag[1])
```

Comment:
might be better to use `Number(splitTag[1])` to parse decimal since quantity can be 2.5